### PR TITLE
help mode: naive forward/reverse text search

### PR DESCRIPTION
**Pull request by csboling**
#### What does this PR do?

Adds the ability to search the full contents of the help mode text, using `C-f` or `C-s` to open a line editor for the search string (forward search, `>` prompt) or `C-r` to search backward (`<` prompt) from the current position. 

#### Provide links to any related discussion on [lines](https://llllllll.co/).

Not sure if this has been proposed before, been wanting this for a while to get more use out of the in-module help.

#### How should this be manually tested?

Go to help mode (alt-H). C-f or C-s will open a forward search (line editor in the bottom row), starting from the current position. Enter to perform a search or go to the next result. The whole line containing a search hit is highlighted. If the result was not found anywhere between your starting position and the end of all help text (i.e., all pages) you will get `xxx: NOT FOUND` on the second-from-bottom row. You can repeat this with a reverse search (C-r) again starting from your current position.

* [x] updated `CHANGELOG.md`
* [x] updated the documentation
* [x] run `make format` on each commit - I thiiink? keep having problems with this one, sorry

### CODE REVIEW

**Comment by scanner-darkly** [comment]
really cool idea!

**Comment by csboling** [comment]
Some sequences of searches can cause a [crash](https://llllllll.co/t/teletype-3-feature-requests-and-discussion/16219/373), looking into this now.

**Comment by tehn** [comment]
happy to merge this when you're ready, and i can post a new release as well!

File: module/help_mode.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "help_mode.h"
+#include "help_mode.h"1
```
Comment (scanner-darkly) [inline module/help_mode.c]: a stray `1` at the end of the line here?

Comment (csboling) [reply module/help_mode.c]: Huh! Seems odd that this did not get caught by CI?

Comment (scanner-darkly) [reply module/help_mode.c]: yeah, strange!

**Pull request merged**